provider/dynamodb.go

```diff
@@ -67,20 +113,37 @@ func dynamodbOnlineStoreFactory(serialized pc.SerializedConfig) (Provider, error
 }
 
 func NewDynamodbOnlineStore(options *pc.DynamodbConfig) (*dynamodbOnlineStore, error) {
-	config := &aws.Config{
-		Region:      aws.String(options.Region),
-		Credentials: credentials.NewStaticCredentials(options.AccessKey, options.SecretKey, ""),
+	args := []func(*config.LoadOptions) error{
+		config.WithRegion(options.Region),
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(options.AccessKey, options.SecretKey, "")),
+	}
+	// If we are using a custom endpoint, such as when running localstack, we should point at it. We'd never set this when
+	// directly accessing DynamoDB on AWS.
+	if options.Endpoint != "" {
+		args = append(args,
+			config.WithEndpointResolver(aws.EndpointResolverFunc(func(service, region string) (aws.Endpoint, error) {
```

Comment:
fyi `WithEndpointResolver` is deprecated; it says to use `WithEndpointResolverWithOptions`

Reply:
Thank you!